src/stream-tcp.c

```diff
@@ -4792,10 +4834,15 @@ int StreamTcpPacket (ThreadVars *tv, Packet *p, StreamTcpThread *stt,
     /* track TCP flags */
     if (ssn != NULL) {
         ssn->tcp_packet_flags |= p->tcph->th_flags;
-        if (PKT_IS_TOSERVER(p))
+        if (PKT_IS_TOSERVER(p)) {
             ssn->client.tcp_flags |= p->tcph->th_flags;
-        else if (PKT_IS_TOCLIENT(p))
+            if (unlikely(ssn->client.tcp_init_flags == 0))
```

Comment:
how can this happen? If we have a ssn we should have run the code first setting the flags?

Reply:
If the first packet seen was TOSERVER then we would've initialized that direction, but not this direction (Technically only relevant if https://github.com/OISF/suricata/pull/5802 is accepted, I guess)